core/arch/arm/kernel/user_ta.c

```diff
@@ -869,25 +869,22 @@ static TEE_Result load_elf_from_store(const TEE_UUID *uuid,
 /* Loads a single ELF file (main executable or library) */
 static TEE_Result load_elf(const TEE_UUID *uuid, struct user_ta_ctx *utc)
 {
-	TEE_Result res;
+	TEE_Result res = TEE_ERROR_ITEM_NOT_FOUND;
 	const struct user_ta_store_ops *op = NULL;
 
 	SCATTERED_ARRAY_FOREACH(op, ta_stores, struct user_ta_store_ops) {
 		DMSG("Lookup user TA ELF %pUl (%s)", (void *)uuid,
 		     op->description);
 
 		res = load_elf_from_store(uuid, op, utc);
-		if (res == TEE_ERROR_ITEM_NOT_FOUND)
-			continue;
-		if (res) {
-			DMSG("res=0x%x", res);
+		DMSG("res=0x%x", res);
+		if (res == TEE_ERROR_ITEM_NOT_FOUND ||
+		    res == TEE_ERROR_STORAGE_NOT_AVAILABLE)
 			continue;
-		}
-
 		return res;
 	}
 
-	return TEE_ERROR_ITEM_NOT_FOUND;
+	return res;
```

Comment:
This may return `TEE_ERROR_ITEM_NOT_FOUND` or `TEE_ERROR_STORAGE_NOT_AVAILABLE` according to last failed attempt (if one of these return code).
I think this should always return `TEE_ERROR_ITEM_NOT_FOUND` in both cases.

Reply:
Agree

Reply:
OK, I'll fix that.